.pre-commit-config.yaml

```diff
@@ -0,0 +1,15 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 22.10.0
+    hooks:
+    -   id: black
+# -   repo: https://github.com/PyCQA/flake8
+#     rev: 5.0.4
+#     hooks:
+#     -   id: flake8
```

Comment:
Just a thought but could add pylint too it's more comprehensive than flake8 (though it is slower) but obviously if flake8 was giving you too many issues don't add it rn

Reply:
Another option is ruff, which from my experience seems to be extremely fast in comparison to the other two.

Reply:
That is true but I know it isn't as comprehensive as pylint

Reply:
I'll make a separate issue for it, I just wanted black as I have a couple other issues I'm working on and want the pre-commits in first. Fixing the linter issues seems like quite a job so I'm leaving it to another issue, I'll update the comments though